packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { Consumer } from 'xfc';
+import styles from './EmbeddedContentConsumer.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The source URL of the content to load.
+   */
+  src: PropTypes.string.isRequired,
+  /**
+   * Notifies the component that the container has been launched.
+   */
+  onLaunch: PropTypes.func,
+  /**
+   * Notifies the component that the container has been authorized.
+   */
+  onAuthorize: PropTypes.func,
+  /**
+   * The component can be configured with an authorization secret.
+   * secret - The authorization secret to be used if the embedded application does not know which domain to trust.
+   */
+  options: PropTypes.shape({
+    secret: PropTypes.string,
+  }),
+  /**
+  * A set of event handlers keyed by the event name.
+  */
+  eventHandlers: PropTypes.arrayOf(PropTypes.shape({
+    key: PropTypes.string,
+    handler: PropTypes.func,
```

Comment:
maybe some docs/comments needed for this custom event handler: One thing I'd like to call out is that `this` will refer to `this.xfcFrame` of EmbeddedContentConsumer by default unless you bind it with something else.

Reply:
Wouldn't it be the responsibility of the component providing the handler to bind this to its scope? @tbiethman thoughts?